src/test/HostActivationTests/NativeHosting/HostContext.cs

```diff
@@ -224,166 +160,368 @@ public void GetDelegate_Multiple(string checkProperties)
+                    .EnvironmentVariable("TEST_SIGNAL_MOCK_EXECUTE_ASSEMBLY", $"{sharedState.AppPath}.signal")
+                    .Execute();
+            }
+            finally
+            {
+                File.Delete(propertyCompatConfig);
+            }
+
+            result.Should().Pass();
+            switch (scenario)
+            {
+                case Scenario.ConfigMultiple:
                     result.Should()
-                        .GetRuntimePropertiesIncludes(LogPrefix.App, sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.GetRuntimePropertiesIncludes(LogPrefix.Secondary, sharedState.ConfigPropertyName, sharedState.ConfigPropertyValue)
-                        .And.GetRuntimePropertiesExcludes(LogPrefix.Secondary, sharedState.AppPropertyName)
-                        .And.HavePropertyMock(sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.NotHavePropertyMock(sharedState.ConfigPropertyName);
+                        .InitializeContextForConfig(appOrConfigPath)
+                        .And.CreateDelegateMock();
                     break;
-                case CheckProperties.GetActive:
+                case Scenario.Mixed:
                     result.Should()
-                        .FailToGetRuntimePropertyValue(LogPrefix.App, sharedState.AppPropertyName, HostInvalidState)
-                        .And.FailToGetRuntimePropertyValue(LogPrefix.App, sharedState.ConfigPropertyName, HostInvalidState)
-                        .And.GetRuntimePropertyValue(LogPrefix.Secondary, sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.FailToGetRuntimePropertyValue(LogPrefix.Secondary, sharedState.ConfigPropertyName, HostPropertyNotFound)
-                        .And.HavePropertyMock(sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.NotHavePropertyMock(sharedState.ConfigPropertyName);
+                        .InitializeContextForApp(appOrConfigPath)
+                        .And.ExecuteAssemblyMock(appOrConfigPath, new string[0]);
                     break;
-                case CheckProperties.GetAllActive:
+                case Scenario.NonContextMixed:
                     result.Should()
-                        .FailToGetRuntimeProperties(LogPrefix.App, HostInvalidState)
-                        .And.GetRuntimePropertiesIncludes(LogPrefix.Secondary, sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.GetRuntimePropertiesExcludes(LogPrefix.Secondary, sharedState.ConfigPropertyName)
-                        .And.HavePropertyMock(sharedState.AppPropertyName, sharedState.AppPropertyValue)
-                        .And.NotHavePropertyMock(sharedState.ConfigPropertyName);
+                        .ExecuteAssemblyMock(appOrConfigPath, new string[0]);
                     break;
```

Comment:
This case should also create a delegeate mock...

Reply:
Actually all of these cases should also create the second context delegate mock... ideally we would ask for a different delegate and verify that it got created...
This also applies to the framework test above (but inly in the success cases)

Reply:
The native host is now always using the `com_activation` type when getting a delegate with the active context and `load_in_memory_assembly` with the secondary context (`CreateDelegateMock_COM/InMemoryAssembly` check the args the mock coreclr prints).